### PR TITLE
mattermost: 8.1.4 -> 8.1.7

### PR DESCRIPTION
**Pull request by mgdelacroix**
## Description of changes

This PR moves Mattermost forward to version `8.1.7`, changelog [can be found here](https://docs.mattermost.com/deploy/legacy-self-hosted-changelog.html#release-v8-1-extended-support-release)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by numinit** [comment]
It will be better to go to 8.1.6 to stay on the Extended Support Release:

https://docs.mattermost.com/deploy/legacy-self-hosted-changelog.html#release-v8-1-extended-support-release

File: pkgs/servers/mattermost/default.nix [hunk 1]
```diff
@@ -8,33 +8,39 @@
...
   subPackages = [ "cmd/mattermost" ];
 
+  tags = [ "production" ];
+
   ldflags = [
     "-s"
     "-w"
     "-X github.com/mattermost/mattermost/server/public/model.Version=${version}"
     "-X github.com/mattermost/mattermost/server/public/model.BuildNumber=${version}-nixpkgs"
-    "-X github.com/mattermost/mattermost/server/public/model.BuildDate=1970-01-01"
```
Comment (numinit) [inline pkgs/servers/mattermost/default.nix:34]: We probably should not change these.

**Review by numinit — CHANGES_REQUESTED** [review]

File: pkgs/servers/mattermost/default.nix (hunk 1 above)
Comment (mgdelacroix) [reply pkgs/servers/mattermost/default.nix:34]: Updated. I left the `BuildHashEnterprise=none` as the generated binary doesn't contain the enterprise part of the codebase

Comment (numinit) [reply pkgs/servers/mattermost/default.nix:34]: Thanks Miguel! I appreciate having another maintainer.

**Review by numinit — APPROVED** [review]

**Comment by mgdelacroix** [comment]
Thanks Morgan for the review and the guidance!

**Review by numinit — APPROVED** [review]

**Comment by mgdelacroix** [comment]
Apologies for the re-review ask @numinit, I was updating the PR to solve the conflicts and bumped the version to `v8.1.7`, which was released yesterday

**Review by numinit — APPROVED** [review]
No worries

**Comment by numinit** [comment]
FWIW I think there is a case for backporting this one since it's a security release.

**Pull request merged**